src/components/Header.jsx

```diff
@@ -1,99 +1,198 @@
 import { Link } from 'react-router-dom';
+
 import '../styles/Header.css'
+
 import { useEffect, useState } from 'react';
 
+
+
 function Header() {
+
     const [isScrolled, setIsScrolled] = useState(false);
+
     const [activePage, setActivePage] = useState('/');
+
     const [isMenuOpen, setIsMenuOpen] = useState(false);
 
+
+
     useEffect(() => {
+
         const handleScroll = () => {
+
             const currentScrollY = window.scrollY
 
+
+
             if (currentScrollY > 50) {
+
                 setIsScrolled(true);
+
             } else {
+
                 setIsScrolled(false);
+
             }
+
         }
+
         window.addEventListener('scroll', handleScroll);
+
         return () => window.removeEventListener('scroll', handleScroll);
+
     }, []);
 
+
+
     const handleLinkClick = (page) => {
+
         setActivePage(page);
+
         setIsMenuOpen(false);
+
     };
 
+
+
     const toggleMenu = () => {
+
         setIsMenuOpen(!isMenuOpen);
+
     };
 
 
+
+
+
     return (
+
         <>
+
             <header className={`header ${isScrolled ? 'hidden' : ''}`}>
+
                 <nav>
+
                     <ul>
+
                         <li><Link to='/' 
+
                         className={activePage === '/' ? 'active' : ''} 
+
                         onClick={() => handleLinkClick('/')}>
+
                         Home</Link></li>
 
+
+
                         <li><Link to='/Research' 
+
                         className={activePage === '/Research' ? 'active' : ''} 
+
                         onClick={() => handleLinkClick('/Research')}>
+
                         Research</Link></li>
 
+
+
                         <li><Link to='/Publications' 
+
                         className={activePage === '/Publications' ? 'active' : ''} 
+
                         onClick={() => handleLinkClick('/Publications')}>
+
                         Publications</Link></li>
 
+
+
                         <li><Link to='/Team' 
+
                         className={activePage === '/Team' ? 'active' : ''} 
+
                         onClick={() => handleLinkClick('/Team')}>
+
                         Team</Link></li>
```

Comment:
Use a map, 

```suggestion
const PAGES = ['publications', ...]
...
{PAGES.map(i => <li key={i}><Link to={`/${i}`} ..... )} 
```

Make sure to add the key to to the higest component

Reply:
Just render different class name is shown or not not an entire new header tag